vespalib/src/vespa/vespalib/util/atomic.h

```diff
@@ -0,0 +1,155 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <atomic>
+#include <type_traits>
+
+/**
+ * Utility functions for single value atomic memory accesses.
+ *
+ * store/load_ref_* functions can be used to provide well-defined atomic
+ * memory access to memory locations that aren't explicitly wrapped in std::atomic
+ * objects. In this case, all potentially racing loads/stores _must_ be through
+ * atomic utility functions (or atomic_ref).
+ *
+ * Non-ref store/load_* functions are just syntactic sugar to make code using
+ * atomics more readable, but additionally adds sanity checks that all atomics
+ * are always lock-free.
+ */
+
+namespace vespalib::atomic {
+
+//
+// std::atomic_ref<T> helpers
+//
+
+// No atomic_ref on clang on darwin (for now)
+#if defined(__clang__) && defined(__apple_build_version__)
+#  define VESPA_NO_ATOMIC_REF_SUPPORT
```

Comment:
Consider `#include <version>` earlier in the file, and use `#if __cpp_lib_atomic_ref` instead of `#ifndef VESPA_NO_ATOMIC_REF_SUPPORT`

Reply:
Neat! That's definitely better, will replace